test/test_torch.py

```diff
@@ -5490,12 +5493,15 @@ def test_put_empty(self, device):
                     src = torch.randn(indices_shape, device=device)
                     self.assertEqual(dst, dst.put_(indices, src, accumulate=accumulate))
 
+    def scatter_allow_reduce(self, device, dtype, reduceop):
+        return device == 'cpu' or (device == 'cuda' and reduceop == 'multiply' and not dtype.is_floating_point)
```

Comment:
The device type needs to be acquired first here:

```
device_type = torch.device(device).type
return device_type == 'cpu' ...
```

This is because the test suite supports running on specific CUDA devices, so the `device` argument to a test might be "cuda:0" or "cuda:1", not "cuda".